colander/__init__.py

```diff
@@ -609,14 +609,26 @@ def _luhnok(value):
     return checksum
 
 
+# Gingerly lifted from Django 1.3.x:
+# https://github.com/django/django/blob/stable/1.3.x/django/core/validators.py#L45
+# <3 y'all!
 URL_REGEX = (
-    r'(?i)\b((?:[a-z][\w-]+:(?:/{1,3}|[a-z0-9%])|www\d{0,3}[.]|'
-    r'[a-z0-9.\-]+[.][a-z]{2,4}/)(?:[^\s()<>]+|\(([^\s()<>]+|'
-    r'(\([^\s()<>]+\)))*\))+(?:\(([^\s()<>]+|(\([^\s()<>]+\)))*\)|'
-    r'[^\s`!()\[\]{};:\'".,<>?«»“”‘’]))'  # "emacs!
+    # {http,ftp}s:// (not required)
+    r'^((?:http|ftp)s?://)?'
+    # Domain
+    r'(?:(?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+'
+    r'(?:[A-Z]{2,6}\.?|[A-Z0-9-]{2,}\.?)|'
+    # Localhost
+    r'localhost|'
+    # IP address?
+    r'\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})'
+    # Optional port
+    r'(?::\d+)?'
+    # Path
+    r'(?:/?|[/?]\S+)$'
```

Comment:
As far as undocumented changes this looks to at least drop support for file urls and other schemes as well as maybe ipv6 addresses?

Reply:
Ugh... ipv6 addresses.

Reply:
Yeah, this does drop file:/// support, which is a uri, not a url.

Reply:
My point was that you left a lot of things undocumented in the changelog at the very least about what used to be accepted versus now. You should also link to the spec to defend the change.

Reply:
I’m not disagreeing. I didn’t realize the old allowed file:/// until you mentioned it. 

I’ll try to add rudimentary ipv6 support back.